baybe/acquisition/base.py

```diff
@@ -81,8 +87,25 @@ def to_botorch(
                 self.get_integration_points(searchspace)  # type: ignore[attr-defined]
             )
 
-        params_dict.update(additional_params)
+        # Add acquisition objective
+        match objective:
+            case SingleTargetObjective(NumericalTarget(mode=TargetMode.MIN)):
+                if issubclass(acqf_cls, bacqf.AnalyticAcquisitionFunction):
+                    additional_params["maximize"] = False
+                elif issubclass(acqf_cls, bacqf.MCAcquisitionFunction):
+                    additional_params["objective"] = LinearMCObjective(
+                        torch.tensor([-1.0])
+                    )
+                else:
+                    raise ValueError(
+                        f"Unsupported acquisition function type: {acqf_cls}."
+                    )
+            case SingleTargetObjective() | DesirabilityObjective():
```

Comment:
My random guess: you haven't yet looked into how `match` works in python, right? I suspect this because I also had to familiarize myself lately and would have asked the same question 😄 Spoiler: python interprets syntax differently than in regular contexts. So chances are high that what you think it does is different from what it actually does 🙃 

I'll let you do some reading first and then ask you to either confirm or raise your question again 👯

Reply:
Ok just had a read about that. That is pretty cool stuff.